# First attempt in cleaning Kaggle Data Set

Kaggle data set provided here: https://www.kaggle.com/ankitkalauni/the-food-delivery-time-for-different-cuisines <br>
Selected based on topic and the 5.9 usability rating.

In [1]:
# imports
import pandas as pd
import numpy as np
import matplotlib as plt
import seaborn as sn

In [2]:
df = pd.read_excel("../Data/Data_Train.xlsx")
df.head()

Restaurant                             Location  \
0    ID_6321  FTI College, Law College Road, Pune   
1    ID_2882                 Sector 3, Marathalli   
2    ID_1595                       Mumbai Central   
3    ID_5929                      Sector 1, Noida   
4    ID_6123   Rmz Centennial, I Gate, Whitefield   

                                 Cuisines Average_Cost Minimum_Order Rating  \
0  Fast Food, Rolls, Burger, Salad, Wraps         ₹200           ₹50    3.5   
1                     Ice Cream, Desserts         ₹100           ₹50    3.5   
2         Italian, Street Food, Fast Food         ₹150           ₹50    3.6   
3          Mughlai, North Indian, Chinese         ₹250           ₹99    3.7   
4                         Cafe, Beverages         ₹200           ₹99    3.2   

  Votes Reviews Delivery_Time  
0    12       4    30 minutes  
1    11       4    30 minutes  
2    99      30    65 minutes  
3   176      95    30 minutes  
4   521     235    65 minutes

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11094 entries, 0 to 11093
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Restaurant     11094 non-null  object
 1   Location       11094 non-null  object
 2   Cuisines       11094 non-null  object
 3   Average_Cost   11094 non-null  object
 4   Minimum_Order  11094 non-null  object
 5   Rating         11094 non-null  object
 6   Votes          11094 non-null  object
 7   Reviews        11094 non-null  object
 8   Delivery_Time  11094 non-null  object
dtypes: object(9)
memory usage: 780.2+ KB


No missing values from the information seen above. <br>
Each column is listed as objects but the last 6 columns (Average_Cost, Minimum_Order, Rating, Votes, Reviews, Delivery_Time) should be integers.

In [16]:
df.columns

Index(['Restaurant', 'Location', 'Cuisines', 'Average_Cost', 'Minimum_Order',
       'Rating', 'Votes', 'Reviews', 'Delivery_Time'],
      dtype='object')

### Average Cost clean up

In [4]:
df.Average_Cost = df.Average_Cost.apply(lambda x: x.replace("₹","").replace(",","")).copy()

In [5]:
df.Average_Cost.unique()
# for?

array(['200', '100', '150', '250', '650', '350', '800', '50', '400',
       '600', '300', '750', '450', '550', '1000', '500', '900', '1200',
       '950', '850', '700', '1150', 'for', '1100', '1400', '2050'],
      dtype=object)

In [7]:
df.loc[df["Average_Cost"] == "for"]

Restaurant         Location   Cuisines Average_Cost Minimum_Order Rating  \
6297    ID_6472  Pune University  Fast Food          for           ₹50    NEW   

     Votes Reviews Delivery_Time  
6297     -       -    30 minutes

In [8]:
# Removed the row containing "for" in the Average Cost column
df.drop(6297,inplace=True)

In [9]:
df.Average_Cost = df.Average_Cost.astype(int)

In [10]:
# Check that the column is cleaned.Looks good
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11093 entries, 0 to 11093
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Restaurant     11093 non-null  object
 1   Location       11093 non-null  object
 2   Cuisines       11093 non-null  object
 3   Average_Cost   11093 non-null  int32 
 4   Minimum_Order  11093 non-null  object
 5   Rating         11093 non-null  object
 6   Votes          11093 non-null  object
 7   Reviews        11093 non-null  object
 8   Delivery_Time  11093 non-null  object
dtypes: int32(1), object(8)
memory usage: 823.3+ KB


### Minimum Order clean up

In [13]:
df.Minimum_Order = df.Minimum_Order.apply(lambda x: x.replace("₹","").replace(",","")).astype(int).copy()

In [14]:
df.Minimum_Order.unique()

array([ 50,  99,   0, 200, 450, 350,  79, 400, 199, 500, 250, 150,  90,
       299, 300, 240,  89,  59])

In [15]:
# Check that the column is cleaned. Looks good
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11093 entries, 0 to 11093
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Restaurant     11093 non-null  object
 1   Location       11093 non-null  object
 2   Cuisines       11093 non-null  object
 3   Average_Cost   11093 non-null  int32 
 4   Minimum_Order  11093 non-null  int32 
 5   Rating         11093 non-null  object
 6   Votes          11093 non-null  object
 7   Reviews        11093 non-null  object
 8   Delivery_Time  11093 non-null  object
dtypes: int32(2), object(7)
memory usage: 780.0+ KB


### Rating clean up

In [19]:
df.Rating.unique()

array(['3.5', '3.6', '3.7', '3.2', '3.8', '4.0', '3.9', '4.2', '-', '2.8',
       '3.0', '3.3', '3.1', '4.7', '3.4', '4.1', 'NEW', '2.9', '4.6',
       '4.3', '2.6', '4.5', '4.4', '4.8', '2.4', '2.7', '2.5',
       'Opening Soon', '2.2', '4.9', '2.3', '2.1', 'Temporarily Closed'],
      dtype=object)

Rating has business hour information as well. For this clean up I will remove the rows with -, Opening Soon, Temporarily Closed and NEW.

In [27]:
# During first check it can be seen that the - is being used in place of no information.
print(df.loc[df["Rating"] == "-"].shape)
print(df.loc[df["Votes"] == "-"].shape)
df.loc[df["Reviews"] == "-"].shape

(1191, 9)
(2073, 9)


(2311, 9)

Each of these is missing different amounts of information. For this I will create a new dataframe and remove everything missing (2311 rows) while the other dataframe will be used to keep order information since that is not missing.